apps/ios/LiveVideo/LiveVideo/Views/Stream/SpeakerVideoViewModelFactory.swift

```diff
@@ -0,0 +1,23 @@
+//
+//  Copyright (C) 2022 Twilio, Inc.
+//
+
+import Foundation
+
+class SpeakerVideoViewModelFactory {
```

Comment:
This code used to be in `SpeakerGridViewModel` but I moved it out so that it could also be used by `PresentationLayoutViewModel`.

If this was a production app it would probably make sense to create an entirely new layer that is above `RoomManager` and below the view models. Such layer would be responsible for merging state from the video SDK and the Sync SDK into a single view of the speakers. But if this was a production app there would probably be a lot more state than just `host` to merge with the video room data. I decided against adding the extra layer because it would add quite a bit of complexity and may be harder for customers to follow. The new layer would probably need to maintain state for all the speakers and have publishers to send out updates.